src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -136,7 +134,8 @@ public async Task<(bool, FunctionMetadataResponse)> TryGetFunction(string name,
         {
             // TODO: DI (FACAVAL) Follow up with ahmels - Since loading of function metadata is no longer tied to the script host, we
             // should be able to inject an IFunctionMedatadaManager here and bypass this step.
-            var functionMetadata = FunctionMetadataManager.ReadFunctionMetadata(Path.Combine(_config.RootScriptPath, name), null, new Dictionary<string, ICollection<string>>(), fileSystem: FileUtility.Instance);
+            var workerConfigs = ScriptHost.GetWorkerConfigs("TODO", _logger);
+            var functionMetadata = FunctionMetadataManager.ReadFunctionMetadata(Path.Combine(_config.RootScriptPath, name), null, workerConfigs, new Dictionary<string, ICollection<string>>(), fileSystem: FileUtility.Instance);
```

Comment:
Rather than pass in the loose collection of worker configs, perhaps it would be cleaner to pass in an ILanguageWorkerService service that provides services like extension => language mappings, etc.? That would allow us to keep all such logic centralized.

Reply:
Don't really need DI - I'm suggesting that you encapsulate these new operations in a simple service implementation in this change set and pass that around, rather than making WorkerConfig public and passing those around.